addon/components/upf-icon.js

```diff
@@ -0,0 +1,49 @@
+import Ember from 'ember';
+
+const {
+  Component,
+  computed,
+  inject
+} = Ember;
+
+const UpfIconComponent = Component.extend({
+  assetMap: inject.service('asset-map'),
+
+  icon: computed('params.[]', function() {
+    return this.get('params')[0];
+  }),
+
+  color: computed('params.[]', function() {
+    return this.get('params')[1];
+  }),
+
+  thickness: computed('params.[]', function() {
+    return this.get('params')[2];
+  }),
+
+  didInsertElement: function() {
+    $.get(
+      this.get('assetMap').resolve(`assets/upf-icons/${this.get('icon')}.svg`),
+      function(data) {
+        let element = this.$()[0];
```

Comment:
I'm not sure, but instead of `this.$()[0]` you can get the component element with `this.element` but need to be tried :)